pulp_service/pulp_service/app/tasks/package_scan.py

```diff
@@ -0,0 +1,88 @@
+import aiohttp
+import asyncio
+import json
+
+from asgiref.sync import sync_to_async
+from queue import Queue
+
+from pulpcore.plugin.models import RepositoryVersion
+from pulp_npm.app.models import Package as NPMPackage
+from pulp_service.app.constants import PKG_ECOSYSTEM, OSV_QUERY_URL
+from pulp_service.app.models import VulnerabilityReport
+
+# Create a thread-safe queue to share Content units between threads
+content_queue = Queue()
+
+
+async def check_content(repo_version_pk):
+    """
+    Get the list of contents from reop_version, build a package_list and makes an API request to
+    osv.dev using this package_list.
+    """
+    # create a thread to collect the Contents from RepositoryVersion
+    await asyncio.to_thread(_get_content_from_repo, repo_version_pk)
+    await _scan_packages()
+
+
+async def _scan_packages():
+    """
+    Makes a request to the osv.dev API and store the results in VulnerabilityReport model.
+    """
+    scanned_packages = {}
+    async with aiohttp.ClientSession() as session:
+        for package in _define_osv_package_list():
```

Comment:
This could be simplified some if you instead of using `_define_osv_package_list()` as a helper function you use `iter` like this. Give this a try:

`for package in iter(content_queue.get, None):`

You can see some about how this works with in [this SO post](https://stackoverflow.com/questions/21157739/how-to-iterate-through-a-python-queue-queue-with-a-for-loop-instead-of-a-while-l).

I would then turn `_define_osv_package_list()` into a helper function which just helps build the `osv_data` object you use in the request. `_define_osv_package_list` already does just, this would just remove the queue fetching part. Also with it's changing role I'd probably rename to `_build_osv_data` or something. Maybe `_define_osv_package_list` is right. Totally up to you.